action.yml

```diff
@@ -50,11 +50,17 @@ runs:
           echo "::error title=⛔ error hint::Support Linux Only"
           exit 1
       - name: Check Auth Info Empty
-        if: ${{ inputs.authkey == '' && (inputs['oauth-secret'] == '' || inputs.tags == '') }}
+        if: ${{ inputs.authkey == '' && inputs['oauth-secret'] == '' }}
         shell: bash
         run: |
           echo "::error title=⛔ error hint::OAuth identity empty, Maybe you need to populate it in the Secrets for your workflow, see more in https://docs.github.com/en/actions/security-guides/encrypted-secrets and https://tailscale.com/s/oauth-clients"
           exit 1
+      - name: Check Tags Provided
+        if: ${{ inputs.authkey == '' && inputs.tags == '' }}
```

Comment:
This Action supports two styles of keys:
- `inputs.authkey = 'tskey-...'` for an original authkey, tied to a User.
- `inputs['oauth-secret'] = '...'` for an OAuth client, from which `tailscaled` will allocate an authkey to use for itself.

A inputs.tags is required if inputs['oauth-secret'] is set. As written, this PR uses `inputs.authkey == ''` to mean "inputs['oauth-secret'] must not be empty or we wouldn't have gotten this far.

I'd suggest instead:
```
        if: ${{ inputs['oauth-secret'] != '' && inputs.tags == '' }}
```